san/async_batch.py

```diff
@@ -0,0 +1,56 @@
+import san.sanbase_graphql
+import asyncio
+from concurrent.futures import ProcessPoolExecutor
+
+from san.sanbase_graphql_helper import QUERY_MAPPING
+from san.query import get_gql_query
+from san.graphql import execute_gql
+from san.transform import transform_query_result
+from san.error import SanError
+
+
+def task(request):
+    [idx, [identifier, kwargs]] = request
+    
+    metric, _separator, slug = identifier.partition("/")
+
+    if metric in QUERY_MAPPING:
+        gql_string_query = get_gql_query(idx, identifier, **kwargs)
+    elif slug != '':
+        gql_string_query = san.sanbase_graphql.get_metric(idx, metric, slug, **kwargs)
+    else:
+        raise SanError('Invalid metric!')
+                
+    response = execute_gql("{" + gql_string_query + "}")
+    return response
+
+class AsyncBatch:
+    def __init__(self):
+        self.queries = []
+
+    def get(self, dataset, **kwargs):
+        self.queries.append([dataset, kwargs])
+
+    def execute(self, max_workers=10):
+        graphql_result = {}
+
+        with ProcessPoolExecutor(max_workers=max_workers) as executor:
```

Comment:
for network requests it is better to use ThreadPoolExecutor, should be less overhead

Reply:
I've changed that.